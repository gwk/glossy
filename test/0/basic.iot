{
  'args': ['-test',
    # single token matches; same as in basic-match.iot.
    ' ',
    'a',
    'bd',
    'bcd',
    'eg',
    'efg',
    'effg',
    'h',
    'hh',
    'ij',
    'k',
    'l',
    # not single token matches; same as in basic-match.iot.
    '',
    'aa',
    'ab',
    'b',
    'bb',
    'bccd',
    'e',
    'ef',
    'g',
    'ah',
    'ha',
    'i',
    'j',
    'ija',
    'kl',
    # multiple-token tests.
    'a bd bcd eg efg effg',
    'h hh ij k l',
    'xx',
    'axa',
  ],
}