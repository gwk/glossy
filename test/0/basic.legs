// Dedicated to the public domain under CC0: https://creativecommons.org/publicdomain/zero/1.0/.

space: \s
a
b_c_opt_d: bc?d
e_f_star_g: ef*g
h_plus: h+
ij

k_or_l: k | l
