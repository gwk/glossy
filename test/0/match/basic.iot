{
  'cmd': 'legs',
  'args': ['test/0/basic.legs', '-match',
    # match.
    ' ',
    'a',
    'bd',
    'bcd',
    'eg',
    'efg',
    'effg',
    'h',
    'hh',
    'ij',
    'k',
    'l',
    # invalid.
    'g',
    'j',
    'x',
    'xx',
    # no match.
    '',
    'aa',
    'ab',
    'b',
    'bb',
    'bccd',
    'e',
    'ef',
    'ah',
    'ha',
    'i',
    'ija',
    'kl',
  ],
}
