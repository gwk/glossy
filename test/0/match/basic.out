match: ' ' -> space
match: 'a' -> a
match: 'bd' -> b_c_opt_d
match: 'bcd' -> b_c_opt_d
match: 'eg' -> e_f_star_g
match: 'efg' -> e_f_star_g
match: 'effg' -> e_f_star_g
match: 'h' -> h_plus
match: 'hh' -> h_plus
match: 'ij' -> ij
match: 'k' -> k_or_l
match: 'l' -> k_or_l
match: 'g' -> invalid
match: 'j' -> invalid
match: 'x' -> invalid
match: 'xx' -> invalid
match: '' -- <none>
match: 'aa' -- <none>
match: 'ab' -- <none>
match: 'b' -- <none>
match: 'bb' -- <none>
match: 'bccd' -- <none>
match: 'e' -- <none>
match: 'ef' -- <none>
match: 'ah' -- <none>
match: 'ha' -- <none>
match: 'i' -- <none>
match: 'ija' -- <none>
match: 'kl' -- <none>
